src/index.ts

```diff
@@ -420,19 +422,48 @@ class HashinalsWalletConnectSDK {
     };
   }
 
+  
+  public async generatePrivateAndPublicKey(): Promise<{
+    privateKey: string;
+    publicKey: string;
+  }> {
+    this.ensureInitialized();
+    const privateKey = await PrivateKey.generateED25519Async();
+    const publicKey = privateKey.publicKey;
+    return {
+      privateKey: privateKey.toString(),
+      publicKey: publicKey.toString()
+    };
+  }
+
+  public async updateTopic(topicId: string, memo: string, adminKey: string): Promise<string> {
+     this.ensureInitialized();
+    let transaction = new TopicUpdateTransaction()
+      .setTopicId(TopicId.fromString(topicId))
+      .setTopicMemo(memo || "")
+      .freezeWith(client);
+      
+    // Convert the adminKey string back to a PrivateKey object
+    const privateKey = PrivateKey.fromString(adminKey);
+    const signedTx = await transaction.sign(privateKey);
+    
+    const receipt = await this.executeTransaction(signedTx);
+    return receipt.topicId!.toString();
+  }
+
   public async createTopic(
     memo?: string,
     adminKey?: string,
     submitKey?: string
   ): Promise<string> {
     this.ensureInitialized();
-
-    let transaction = new TopicCreateTransaction().setTopicMemo(memo || '');
+    let transaction = new TopicCreateTransaction().setTopicMemo(memo || "");
 
     if (adminKey) {
       const adminWithPrivateKey = PrivateKey.fromString(adminKey);
       transaction.setAdminKey(adminWithPrivateKey.publicKey);
-      transaction = await transaction.sign(adminWithPrivateKey);
+      transaction.freezeWith(client); // Freeze after setting the admin key
+      transaction = await transaction.sign(adminWithPrivateKey); // Then sign
```

Comment:
It looks like you're not disabling the signing in `executeTransaction`, so ultimately this transaction will still fail because its already frozen